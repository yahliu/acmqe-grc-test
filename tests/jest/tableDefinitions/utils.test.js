/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'
import {
  createComplianceLink,
  getControls,
  getStandards,
  getCategories,
  getDecisionCount,
  formatAnnotationString,
  getPolicyCompliantStatus
} from '../../../src-web/tableDefinitions/utils'
import {
  disabledPolicyCompliantStatusItem,
  unavailableClusterCompliant,
  withClusterViolation,
  withoutClusterViolation,
  withUnknownClusterViolation
 } from './DefinitionsTestingData'

describe('tableDefinitions utils - createComplianceLink', () => {
  it('should return valid compliance link', () => {
    const item1 = {
      metadata: {
        name: 'testCreateComplianceLink1',
        namespace: 'testNamespace1'
      }
    }
    const param = ['param1', 'param2', 'param3']
    expect(createComplianceLink(item1, ...param)).toMatchSnapshot()

    const item2 = {
      metadata: {
        name: 'testCreateComplianceLink2',
        namespace: 'testNamespace2'
      },
      raw : {
        kind: 'Compliance'
      }
    }
    expect(createComplianceLink(item2)).toMatchSnapshot()

    const item3 = {
      metadata: {
        name: 'testCreateComplianceLink3',
        namespace: 'testNamespace3'
      },
      raw : {
        kind: 'Default'
      }
    }
    expect(createComplianceLink(item3)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - getControls', () => {
  it('should return control status', () => {
    const items = {
      metadata: {
        name: 'testGetControls',
        annotations: {
          'policy.open-cluster-management.io/controls': '1,2,3,4,5'
        }
      },
    }
    expect(getControls(items)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - getStandards', () => {
  it('should return standards status', () => {
    const items = {
      metadata: {
        name: 'testGetStandards',
        annotations: {
          'policy.open-cluster-management.io/standards': '6,7,8,9,10'
        }
      },
    }
    expect(getStandards(items)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - getCategories', () => {
  it('should return categories status', () => {
    const items = {
      metadata: {
        name: 'testGetCategories',
        annotations: {
          'policy.open-cluster-management.io/categories': '11,12,13,14,15'
        }
      },
    }
    expect(getCategories(items)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - getDecisionCount should return the total', () => {
  it('should return decisions status', () => {
    const items = {
      status: {
        decisions: [
          {clusterName:'clusterName1'},
          {clusterName:'clusterName2'},
          {clusterName:'clusterName3'}
        ]
      },
    }
    expect(getDecisionCount(items)).toMatchSnapshot()
  })
  it('tableDefinitions utils - getDecisionCount should return 0', () => {
    const items = {}
    expect(getDecisionCount(items)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - formatAnnotationString', () => {
  const policy = {
    metadata: {
      annotations: {
        'testKey': 'test1 test2,test3 ,  test4 test5',
      }
    }
  }
  it('should get list of annotations back with proper whitespace', () => {
    expect(formatAnnotationString(policy, 'testKey')).toMatchSnapshot()
  })
  it('should get "-" ', () => {
    const item = ''
    expect(formatAnnotationString(item)).toMatchSnapshot()
  })
  it('should get "-" ', () => {
    const item = null
    expect(formatAnnotationString(item)).toMatchSnapshot()
  })
})

describe('tableDefinitions utils - getPolicyCompliantStatus', () => {
  it('should return disabled policy compliant status', () => {
    expect(getPolicyCompliantStatus(disabledPolicyCompliantStatusItem, 'us')).toMatchSnapshot()
  })

  it('should return policy compliant status with unavailable cluster compliant', () => {
    expect(getPolicyCompliantStatus(unavailableClusterCompliant, 'us')).toMatchSnapshot()
  })

  it('should return policy compliant status with cluster violation', () => {
    expect(getPolicyCompliantStatus(withClusterViolation, 'us')).toMatchSnapshot()
  })

  it('should return policy compliant status without cluster violation', () => {
    expect(getPolicyCompliantStatus(withoutClusterViolation, 'us')).toMatchSnapshot()
  })

  it('should return policy compliant status with unknown cluster violation', () => {
    expect(getPolicyCompliantStatus(withUnknownClusterViolation, 'us')).toMatchSnapshot()
  })
})
