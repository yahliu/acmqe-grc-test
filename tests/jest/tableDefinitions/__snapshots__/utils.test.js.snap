// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableDefinitions utils - createComplianceLink should return valid compliance link 1`] = `"testCreateComplianceLink1"`;

exports[`tableDefinitions utils - createComplianceLink should return valid compliance link 2`] = `
<Link
  to="undefined/all/testCreateComplianceLink2"
>
  testCreateComplianceLink2
   (Deprecated)
</Link>
`;

exports[`tableDefinitions utils - createComplianceLink should return valid compliance link 3`] = `
<Link
  to="undefined/all/testNamespace3/testCreateComplianceLink3"
>
  testCreateComplianceLink3
</Link>
`;

exports[`tableDefinitions utils - formatAnnotationString should get "-"  1`] = `"-"`;

exports[`tableDefinitions utils - formatAnnotationString should get "-"  2`] = `"-"`;

exports[`tableDefinitions utils - formatAnnotationString should get list of annotations back with proper whitespace 1`] = `"test1 test2, test3, test4 test5"`;

exports[`tableDefinitions utils - getCategories should return categories status 1`] = `"11, 12, 13, 14, 15"`;

exports[`tableDefinitions utils - getControls should return control status 1`] = `"1, 2, 3, 4, 5"`;

exports[`tableDefinitions utils - getDecisionCount should return the total should return decisions status 1`] = `3`;

exports[`tableDefinitions utils - getDecisionCount should return the total tableDefinitions utils - getDecisionCount should return 0 1`] = `0`;

exports[`tableDefinitions utils - getPolicyCompliantStatus should return disabled policy compliant status 1`] = `
<div
  className="disabledStatus"
>
  --
</div>
`;

exports[`tableDefinitions utils - getPolicyCompliantStatus should return policy compliant status with cluster violation 1`] = `
<div
  className="violationCell"
>
  <RedExclamationCircleIcon
    tooltip="Policy is non-compliant."
  />
  1/3
</div>
`;

exports[`tableDefinitions utils - getPolicyCompliantStatus should return policy compliant status with unavailable cluster compliant 1`] = `
<div
  className="violationCell"
>
  <YellowExclamationTriangleIcon
    tooltip="Policy status is unavailable for some clusters."
  />
  -
</div>
`;

exports[`tableDefinitions utils - getPolicyCompliantStatus should return policy compliant status with unknown cluster violation 1`] = `
<div
  className="violationCell"
>
  <RedExclamationCircleIcon
    tooltip="Policy is non-compliant."
  />
  <YellowExclamationTriangleIcon
    tooltip="Policy status is unavailable for some clusters."
  />
  1/3
</div>
`;

exports[`tableDefinitions utils - getPolicyCompliantStatus should return policy compliant status without cluster violation 1`] = `
<div
  className="violationCell"
>
  <GreenCheckCircleIcon
    tooltip="Policy is compliant."
  />
  0/3
</div>
`;

exports[`tableDefinitions utils - getStandards should return standards status 1`] = `"6, 7, 8, 9, 10"`;
